File: source/clibCore/Models/Entity/tableModel.vb

Option Explicit On
Option Strict On

Imports System
Imports System.Collections.Generic
Imports System.Text
Imports Contensive.BaseClasses
Imports Contensive.Core.Controllers
Imports Newtonsoft.Json

Namespace Contensive.Core.Models.Entity
    Public Class tableModel
        Inherits baseModel
        '
        '====================================================================================================
        '-- const
        Public Const contentName As String = "tables" '<------ set content name
        Public Const contentTableName As String = "ccTables" '<------ set to tablename for the primary content (used for cache names)
        Private Shadows Const contentDataSource As String = "default" '<----- set to datasource if not default
        '
        '====================================================================================================
        ' -- instance properties
        Public Property DataSourceID As Integer
        '
        '====================================================================================================
        Public Overloads Shared Function add(cpCore As coreClass) As tableModel
            Return add(Of tableModel)(cpCore)
        End Function
        '
        '====================================================================================================
        Public Overloads Shared Function add(cpCore As coreClass, ByRef callersCacheNameList As List(Of String)) As tableModel
            Return add(Of tableModel)(cpCore, callersCacheNameList)
        End Function
        '
        '====================================================================================================
        Public Overloads Shared Function create(cpCore As coreClass, recordId As Integer) As tableModel
            Return create(Of tableModel)(cpCore, recordId)
        End Function
        '
        '====================================================================================================
        Public Overloads Shared Function create(cpCore As coreClass, recordId As Integer, ByRef callersCacheNameList As List(Of String)) As tableModel
            Return create(Of tableModel)(cpCore, recordId, callersCacheNameList)
        End Function
        '
        '====================================================================================================
        Public Overloads Function create(cpCore As coreClass, recordGuid As String) As tableModel
            Return MyBase.create(Of tableModel)(cpCore, recordGuid)
        End Function
        '
        '====================================================================================================
        Public Overloads Function create(cpCore As coreClass, recordGuid As String, ByRef callersCacheNameList As List(Of String)) As tableModel
            Return MyBase.create(Of tableModel)(cpCore, recordGuid, callersCacheNameList)
        End Function
        '
        '====================================================================================================
        Public Overloads Shared Function createByName(cpCore As coreClass, recordName As String) As tableModel
            Return createByName(Of tableModel)(cpCore, recordName)
        End Function
        '
        '====================================================================================================
        Public Overloads Shared Function createByName(cpCore As coreClass, recordName As String, ByRef callersCacheNameList As List(Of String)) As tableModel
            Return createByName(Of tableModel)(cpCore, recordName, callersCacheNameList)
        End Function
        '
        '====================================================================================================
        Public Overloads Sub save(cpCore As coreClass)
            MyBase.save(cpCore)
        End Sub
        '
        '====================================================================================================
        Public Overloads Shared Sub delete(cpCore As coreClass, recordId As Integer)
            delete(Of tableModel)(cpCore, recordId)
        End Sub
        '
        '====================================================================================================
        Public Overloads Shared Sub delete(cpCore As coreClass, ccGuid As String)
            delete(Of tableModel)(cpCore, ccGuid)
        End Sub
        '
        '====================================================================================================
        Public Overloads Shared Function createList(cpCore As coreClass, sqlCriteria As String, callersCacheNameList As List(Of String)) As List(Of tableModel)
            Return createList(Of tableModel)(cpCore, sqlCriteria, callersCacheNameList)
        End Function
        '
        '====================================================================================================
        Public Overloads Shared Function createList(cpCore As coreClass, sqlCriteria As String) As List(Of tableModel)
            Return createList(Of tableModel)(cpCore, sqlCriteria)
        End Function
        '
        '====================================================================================================
        Public Overloads Sub invalidatePrimaryCache(cpCore As coreClass, recordId As Integer)
            invalidatePrimaryCache(Of tableModel)(cpCore, recordId)
        End Sub
        '
        '====================================================================================================
        Public Overloads Shared Function getRecordName(cpcore As coreClass, recordId As Integer) As String
            Return baseModel.getRecordName(Of tableModel)(cpcore, recordId)
        End Function
        '
        '====================================================================================================
        Public Overloads Shared Function getRecordName(cpcore As coreClass, ccGuid As String) As String
            Return baseModel.getRecordName(Of tableModel)(cpcore, ccGuid)
        End Function
        '
        '====================================================================================================
        Public Overloads Shared Function getRecordId(cpcore As coreClass, ccGuid As String) As Integer
            Return baseModel.getRecordId(Of tableModel)(cpcore, ccGuid)
        End Function
        '
        '====================================================================================================
        Public Overloads Shared Function createDefault(cpcore As coreClass) As tableModel
            Return createDefault(Of tableModel)(cpcore)
        End Function
    End Class
End Namespace

File: source/clibCoreCs/Models/Entity/tableModel.cs

using System;
using System.Reflection;
using System.Xml;
using System.Diagnostics;
using System.Linq;
using System.Text.RegularExpressions;
using System.Collections.Generic;
using System.Data;
using System.Data.SqlClient;
using Contensive.Core;
using Contensive.Core.Models.Entity;
using Contensive.Core.Controllers;
using static Contensive.Core.Controllers.genericController;
using static Contensive.Core.constants;
//
namespace Contensive.Core.Models.Entity {
    public class tableModel : baseModel {
        //
        //====================================================================================================
        //-- const
        public const string contentName = "tables"; //<------ set content name
        public const string contentTableName = "ccTables"; //<------ set to tablename for the primary content (used for cache names)
        private const string contentDataSource = "default"; //<----- set to datasource if not default
        //
        //====================================================================================================
        // -- instance properties
        public int DataSourceID { get; set; }
        //
        //====================================================================================================
        public static tableModel add(coreClass cpCore) {
            return add<tableModel>(cpCore);
        }
        //
        //====================================================================================================
        public static tableModel add(coreClass cpCore, ref List<string> callersCacheNameList) {
            return add<tableModel>(cpCore, ref callersCacheNameList);
        }
        //
        //====================================================================================================
        public static tableModel create(coreClass cpCore, int recordId) {
            return create<tableModel>(cpCore, recordId);
        }
        //
        //====================================================================================================
        public static tableModel create(coreClass cpCore, int recordId, ref List<string> callersCacheNameList) {
            return create<tableModel>(cpCore, recordId, ref callersCacheNameList);
        }
        //
        //====================================================================================================
        public static tableModel create(coreClass cpCore, string recordGuid) {
            return create<tableModel>(cpCore, recordGuid);
        }
        //
        //====================================================================================================
        public static tableModel create(coreClass cpCore, string recordGuid, ref List<string> callersCacheNameList) {
            return create<tableModel>(cpCore, recordGuid, ref callersCacheNameList);
        }
        //
        //====================================================================================================
        public static tableModel createByName(coreClass cpCore, string recordName) {
            return createByName<tableModel>(cpCore, recordName);
        }
        //
        //====================================================================================================
        public static tableModel createByName(coreClass cpCore, string recordName, ref List<string> callersCacheNameList) {
            return createByName<tableModel>(cpCore, recordName, ref callersCacheNameList);
        }
        //
        //====================================================================================================
        public void save(coreClass cpCore) {
            base.save(cpCore);
        }
        //
        //====================================================================================================
        public static void delete(coreClass cpCore, int recordId) {
            delete<tableModel>(cpCore, recordId);
        }
        //
        //====================================================================================================
        public static void delete(coreClass cpCore, string ccGuid) {
            delete<tableModel>(cpCore, ccGuid);
        }
        //
        //====================================================================================================
        public static List<tableModel> createList(coreClass cpCore, string sqlCriteria, string sqlOrderBy, List<string> callersCacheNameList) {
            return createList<tableModel>(cpCore, sqlCriteria, sqlOrderBy, callersCacheNameList);
        }
        //
        //====================================================================================================
        public static List<tableModel> createList(coreClass cpCore, string sqlCriteria, string sqlOrderBy) {
            return createList<tableModel>(cpCore, sqlCriteria, sqlOrderBy);
        }
        //
        //====================================================================================================
        public static List<tableModel> createList(coreClass cpCore, string sqlCriteria) {
            return createList<tableModel>(cpCore, sqlCriteria);
        }
        //
        //====================================================================================================
        public void invalidatePrimaryCache(coreClass cpCore, int recordId) {
            invalidateCacheSingleRecord<tableModel>(cpCore, recordId);
        }
        //
        //====================================================================================================
        public static string getRecordName(coreClass cpcore, int recordId) {
            return baseModel.getRecordName<tableModel>(cpcore, recordId);
        }
        //
        //====================================================================================================
        public static string getRecordName(coreClass cpcore, string ccGuid) {
            return baseModel.getRecordName<tableModel>(cpcore, ccGuid);
        }
        //
        //====================================================================================================
        public static int getRecordId(coreClass cpcore, string ccGuid) {
            return baseModel.getRecordId<tableModel>(cpcore, ccGuid);
        }
        //
        //====================================================================================================
        public static tableModel createDefault(coreClass cpcore) {
            return createDefault<tableModel>(cpcore);
        }
    }
}
